teps/0128-scheduled-runs.md

```diff
@@ -0,0 +1,279 @@
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Performance](#performance)
+  - [Security](#security)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+- [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes allowing Task/Pipeline users to schedule runs on a recurring basis.
+
+## Motivation
+
+Currently, to get a run to occur on a regular basis, a user must write a cron job that either creates the run itself
+(which duplicates much of triggers' functionality) or pings an EventListener.
+If the EventListener is exposed outside of the cluster, the user must also sign the cron job event body and verify it in a custom interceptor
+(although it's unlikely a user would choose to expose an ingress for an EventListener used only for cron jobs).
+
+### Goals
+
+- Can create new PipelineRuns, TaskRuns, or CustomRuns on a recurring basis without creating a webhook or writing a CronJob
+
+### Non-Goals
+
+- Polling a repository (or other external system) for changes; this is covered in [TEP-0083](./0083-polling-runs-in-tekton.md).
+
+### Use Cases
+
+- Release Pipeline: User would like to create releases of their project on a nightly basis.
+
+## Proposal
+
+Add a `schedule` field following Kubernetes CronJob syntax to the EventListener CRD, for example:
+
+```yaml
+kind: EventListener
+spec:
+  schedule: "0 0 * * *"
+  triggers:
+  - bindings:
+    - name: project-name
+      value: wait-task
+    - name: triggers-uuid
+      value: $(context.eventID)
+    - name: datetime
+      value: $(context.datetime)
+    template:
+      ref: experimental-release-pipeline-template
+```
+
+```yaml
+kind: EventListener
+spec:
+  schedule: "0 0 * * *"
+  triggerGroups:
+  - triggerSelector:
+      labelSelector:
+        matchLabels:
+          type: nightly-release
+```
+
+An EventListener with a `schedule` would send an empty event body at the specified times.
+This means that any bindings containing `$(body.foo)` or `$(header.bar)` would fail, as well as any interceptors
+responsible for processing the event.
+An EventListener with a `schedule` would still create a service for receiving events.
+
+### Bindings
+
+As part of future work for this proposal, we can add two new variable replacements in bindings:
+- $(context.date): date when the run was created, in RFC3339 format
+- $(context.datetime): timestamp when the run was created, in RFC3339 format
+
+### CLI
+
+As part of future work for this proposal, we can add a CLI command to manually trigger execution of an EventListener,
+similar to the kubectl functionality `kubectl create job --from=cronjob/my-cronjob`, for example:
+
+```sh
+tkn eventlistener start <el_name>
```

Comment:
+1 A more general purpose could be a nice addition too `tkn eventlistener start -t trigger-name  --body ${} ` etc

Reply:
good idea, added this example syntax